services/template.py

```diff
@@ -0,0 +1,15 @@
+from jinja2 import Environment, PackageLoader, StrictUndefined, select_autoescape
+
+
+class TemplateService:
+    def __init__(self):
+        # this loads the templates from the templates directory in this repository since it's looking for a dir named `templates` next to app.py
+        self.env = Environment(
```

Comment:
so, regarding using a singleton here:

https://jinja.palletsprojects.com/en/3.0.x/api/
> Most applications will create one [Environment](https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.Environment) object on application initialization and use that to load templates. In some cases however, it’s useful to have multiple environments side by side, if different configurations are in use.

https://stackoverflow.com/questions/618827/optimizing-jinja2-environment-creation
> My application is running on Google App Engine and most of requests constantly gets yellow flag due to high CPU usage. Using profiler I tracked the issue down to the routine of creating jinja2.Environment instance.

it looks like creating an `Environment` is expensive so we should try to only do it once. either that or we precompile templates with something like https://github.com/MiCHiLU/jinja2-precompiler but that approach doesn't seem well-documented.

i don't have a strong opinion on how to accomplish this. you could make `env` a global or class-level variable instead of an instance variable and be done with it, or you could make `TemplateService` a real singleton again. if you go the singleton route, there's a way to make it a singleton that you still invoke like `TemplateService().get_template()`: https://code.activestate.com/recipes/66531/ (maybe not technically a singleton but splitting hairs)

@giovanni-guidini what were the "cached ENV" issues you alluded to? were those, like, env vars or specifically a `jinja2.Environment`?